---
title: "clustering"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = 'C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp')
```


### Differential gene expression analysis

Load in the data:
```{r}
load('clustered-all.rda')
```

Looking at various contingency tables for different variables. 
```{r}
table(All@meta.data$orig.ident)
table(All@meta.data$orig.ident, All@meta.data$Sex)
table(All@meta.data$orig.ident, All@meta.data$Treat)
table(All@meta.data$orig.ident, All@meta.data$res.0.4)
```


```{r}
library(edgeR)
library(Seurat)
library(EnhancedVolcano)
library(gridExtra)
library(grid)
library(ggplot2)
library(corrplot)
library(UpSetR)
library(gridGraphics)
library(readxl)
library(magrittr)
library(topGO)
library(cowplot)
library(org.Mm.eg.db)
library(openxlsx)
```


Renaming data for easier handling, separating the counts matrix and the cell attribute matrix and the tSNE coordinates for future use.
```{r}
counts_mat = All@assays$RNA@data
rownames(counts_mat) = All@assays$RNA@data@Dimnames[[1]]
colnames(counts_mat) = All@assays$RNA@data@Dimnames[[2]]
cells_mat = All@meta.data 
counts_mat = counts_mat[,rownames(cells_mat)]
```



This is edgeR differential expression for stratifying between cluster assignments. Model matrix designed so that the average effect of the 4 Pb mice minus the average effect of the 4 Ctrl mice would result in a contrast of only the 1*Tx term.
```{r}
calc_fit <- function(clusterID) {
	print(sprintf("Analyzing cluster %s",clusterID))
	#counts_clu0 = counts_mat[,cluster_probs[,sprintf("res.0.4%s",clusterID)]>0.95]
	counts_clu0 = counts_mat[,colnames(counts_mat) %in% rownames(cells_mat)[cells_mat$seurat_clusters %in% clusterID]]
	cells_clu0 = cells_mat[colnames(counts_clu0),]
	#Ctrls: 109d 113c 118d 119g
	#Pbs  : 103c 107d 112c 116d
	cells_clu0$Tx = 0.5*((cells_clu0$Treat == "Pb") - (cells_clu0$Treat == "Ctrl"))
	cells_clu0$TxCtl.M1 = - (cells_clu0$orig.ident == "Ctl.M1") + (cells_clu0$orig.ident == "Ctl.F2")
	cells_clu0$TxCtl.F1 = - (cells_clu0$orig.ident == "Ctl.F1") + (cells_clu0$orig.ident == "Ctl.F2")
	cells_clu0$TxCtl.M2 = - (cells_clu0$orig.ident == "Ctl.M2") + (cells_clu0$orig.ident == "Ctl.F2")
	cells_clu0$TxPb.F1 =  (cells_clu0$orig.ident == "Pb.F1") - (cells_clu0$orig.ident == "Pb.M2")
	cells_clu0$TxPb.M1 =  (cells_clu0$orig.ident == "Pb.M1") - (cells_clu0$orig.ident == "Pb.M2")
	cells_clu0$TxPb.F2 =  (cells_clu0$orig.ident == "Pb.F2") - (cells_clu0$orig.ident == "Pb.M2")
	cells_clu0$det = colSums(as.matrix(counts_clu0 > 0))/ncol(counts_clu0)
	
	design = model.matrix(~ Tx+TxCtl.M1+TxCtl.F1+TxCtl.M2+TxPb.F1+TxPb.M1+TxPb.F2+det, data = cells_clu0)
	y = DGEList(counts_clu0, group = factor(cells_clu0$Treat))
	y = calcNormFactors(y)
	print("Calcuating Dispersion")
	y = estimateDisp(y, design)
	print("Fitting model")
	fit = glmQLFit(y, design)
	print("Calculating QLFTest")
	qlf = glmQLFTest(fit, contrast=c(0,1,0,0,0,0,0,0,0))
	return(qlf)
}
```

This is edgeR DE analysis for all clusters
```{r}
calc_fit_all <- function() {
	print(sprintf("Analyzing cluster %s","ALL"))
	#counts_clu0 = counts_mat[,cluster_probs[,sprintf("res.0.4%s",clusterID)]>0.95]
	counts_clu0 = counts_mat
	cells_clu0 = cells_mat
	
	#Ctrls: 109d 113c 118d 119g
	#Pbs  : 103c 107d 112c 116d
	cells_clu0$Tx = 0.5*((cells_clu0$Treat == "Pb") - (cells_clu0$Treat == "Ctrl"))
	cells_clu0$TxCtl.M1 = - (cells_clu0$orig.ident == "Ctl.M1") + (cells_clu0$orig.ident == "Ctl.F2")
	cells_clu0$TxCtl.F1 = - (cells_clu0$orig.ident == "Ctl.F1") + (cells_clu0$orig.ident == "Ctl.F2")
	cells_clu0$TxCtl.M2 = - (cells_clu0$orig.ident == "Ctl.M2") + (cells_clu0$orig.ident == "Ctl.F2")
	cells_clu0$TxPb.F1 =  (cells_clu0$orig.ident == "Pb.F1") - (cells_clu0$orig.ident == "Pb.M2")
	cells_clu0$TxPb.M1 =  (cells_clu0$orig.ident == "Pb.M1") - (cells_clu0$orig.ident == "Pb.M2")
	cells_clu0$TxPb.F2 =  (cells_clu0$orig.ident == "Pb.F2") - (cells_clu0$orig.ident == "Pb.M2")
	cells_clu0$det = colSums(as.matrix(counts_clu0 > 0))/ncol(counts_clu0)
	
	design = model.matrix(~ Tx+TxCtl.M1+TxCtl.F1+TxCtl.M2+TxPb.F1+TxPb.M1+TxPb.F2+det, data = cells_clu0)
	y = DGEList(counts_clu0, group = factor(cells_clu0$Treat))
	y = calcNormFactors(y)
	print("Calcuating Dispersion")
	y = estimateDisp(y, design)
	print("Fitting model")
	fit = glmQLFit(y, design)
	print("Calculating QLFTest")
	qlf = glmQLFTest(fit, contrast=c(0,1,0,0,0,0,0,0,0))
	return(qlf)
}

qlf <- calc_fit_all()
#saveRDS(qlf, file="Pb.clustALL.fit.rds")
#qlf <- readRDS('Pb.clustALL.fit.rds')
qlf.res <- topTags(qlf,n=nrow(qlf))$table

write.csv(qlf.res$table,file='C:/Users/John/GoogleDrive/Pb_Hipp_Research/Results/Pb_DE_genes_across_all_clusters.csv')


table(qlf.res$table$FDR<0.05)
qlf.fdr <- qlf.res$table %>% filter(FDR<0.05)
table(abs(qlf.fdr$logFC)>0.5)

qlf.res[qlf.res$FDR<0.05 & abs(qlf.res$logFC)>0.5,]


#supplemental figure 4

FeaturePlot(object=All, features = c("Hbb-bs"),col=c("white","blue")) 
FeaturePlot(object=All, features = c("Gm42726"),col=c("white","blue")) 
FeaturePlot(object=All, features = c("Kif5a"),col=c("white","blue")) 
FeaturePlot(object=All, features = c("Hapln2"),col=c("white","blue")) 
FeaturePlot(object=All, features = c("Gm15013"),col=c("white","blue")) 
DimPlot(object = All,reduction='tsne',group.by='seurat_clusters')

#how is it by mouse?
VlnPlot(All, features = "Hbb-bs", point.size.use=0.3, group.by = 'orig.ident') 
VlnPlot(All, features = "Gm42726", point.size.use=0.3, group.by = 'orig.ident') 
VlnPlot(All, features = "Kif5a", point.size.use=0.3, group.by = 'orig.ident') 
VlnPlot(All, features = "Hapln2", point.size.use=0.3, group.by = 'orig.ident') 
VlnPlot(All, features = "Gm15013", point.size.use=0.3, group.by = 'orig.ident') 



### combined figure 2

point.size <- 0.2
text.size <- 14

f2A <- EnhancedVolcano(qlf.res, 
                lab=rownames(qlf.res), 
                title='',
                subtitle='',
                x='logFC', 
                xlim=c(-1,1),
                y='FDR',
                pCutoff = 0.05,
                FCcutoff = 0.5,
                ylab = bquote(~-Log[10]~adjusted~italic(P)),
                transcriptLabSize = 5.2,
                transcriptPointSize = 1.8,
                transcriptLabhjust = 0.4,
                #drawConnectors = T,
                #boxedlabels = T,
                colAlpha = 1,
                legendVisible = F,
                caption="")
f2B <- VlnPlot(All, features = "Hbb-bs", point.size.use=point.size, group.by = 'Treat') + 
  geom_boxplot(alpha=0.7, outlier.shape=NA, show.legend=F) +
  theme(text=element_text(size=text.size), legend.position="none", 
        axis.title.x=element_blank(), axis.text=element_text(size=text.size))
f2C <- VlnPlot(All, features = "Gm42726", point.size.use=point.size, group.by = 'Treat') + 
  geom_boxplot(alpha=0.7, outlier.shape=NA, show.legend=F) +
  theme(text=element_text(size=text.size), legend.position="none", 
        axis.title.x=element_blank(), axis.text=element_text(size=text.size))
f2D <- VlnPlot(All, features = "Kif5a", point.size.use=point.size, group.by = 'Treat') + 
  geom_boxplot(alpha=0.7, outlier.shape=NA, show.legend=F) +
  theme(text=element_text(size=text.size), legend.position="none", 
        axis.title.x=element_blank(), axis.text=element_text(size=text.size))
f2E <- VlnPlot(All, features = "Hapln2", point.size.use=point.size, group.by = 'Treat') + 
  geom_boxplot(alpha=0.7, outlier.shape=NA, show.legend=F) +
  theme(text=element_text(size=text.size), legend.position="none", 
        axis.title.x=element_blank(), axis.text=element_text(size=text.size))
f2F <- VlnPlot(All, features = "Gm15013", point.size.use=point.size, group.by = 'Treat') + 
  geom_boxplot(alpha=0.7, outlier.shape=NA, show.legend=F) +
  theme(text=element_text(size=text.size), legend.position="none", 
        axis.title.x=element_blank(), axis.text=element_text(size=text.size))


fig2_plots <- list(f2A, f2B, f2C, f2D, f2E, f2F)
fig2_label <- list("A.", "B.", "C.", "D.", "E.", "F.")
fig2_plots <- lapply(1:length(fig2_plots), FUN = function(i){
  arrangeGrob(fig2_plots[[i]], top = textGrob(fig2_label[[i]], x= unit(0, "npc")
                                ,y= unit(1, "npc"), just= c("left","top"),
                                gp=gpar(col="black", fontsize=18)))
})

pdf("Figure_2_DE_overall.pdf", height=10, width=10)
grid.arrange(grobs=fig2_plots,
             layout_matrix= rbind(c(1,1,2),
                                  c(1,1,3),
                                  c(4,5,6)))
dev.off()


tiff("Figure_2_DE_overall.tif", height=10, width=10, units="in", res=300)
grid.arrange(grobs=fig2_plots,
             layout_matrix= rbind(c(1,1,2),
                                  c(1,1,3),
                                  c(4,5,6)))
dev.off()


```


Cluster specific results. The clusters with fewer cells should be more cautious in terms of interpreting results. 
```{r}

cluster.specific <- list()


cluster.specific[[1]] <- calc_fit(0)
cluster.specific[[2]] <- calc_fit(1)
cluster.specific[[3]] <- calc_fit(2)
cluster.specific[[4]] <- calc_fit(3)
cluster.specific[[5]] <- calc_fit(4)
cluster.specific[[6]] <- calc_fit(5)
cluster.specific[[7]] <- NULL #cluster.specific[[7]] <- calc_fit(6)
cluster.specific[[8]] <- calc_fit(7)
cluster.specific[[9]] <- calc_fit(8)
cluster.specific[[10]] <- calc_fit(9)
cluster.specific[[11]] <- calc_fit(10)
cluster.specific[[12]] <- calc_fit(11)

#design matrix not full rank in cluster 6, 10, 11

cluster.specific[[11]] <- NULL
cluster.specific[[12]] <- NULL

save(cluster.specific,file='cluster.specific.rda')

```



Explore cluster specific DE genes
```{r}
setwd('C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp')
load('cluster.specific.rda')

clust <- 0

res <- cluster.specific[[clust+1]]
res.top <- topTags(res,n=nrow(res))
head(res.top$table)

```



Gather table of the FDR significant genes per cluster.
```{r}
clust.fdr <- c()

for(i in 1:length(cluster.specific)){
  res <- cluster.specific[[i]]
  print(paste0('cluster ',i-1))
  flush.console()
  if(!is.null(res)){
    res.top <- topTags(res,n=nrow(res))$table
    res.top <- res.top[res.top$FDR<0.05,]
    if(nrow(res.top)!=0){
      print(paste0('has fdr genes'))
      flush.console()
      res.top$gene <- rownames(res.top)
      res.top$clust <- i-1
      rownames(res.top) <- NULL
      clust.fdr <- rbind(clust.fdr,res.top)
    }
  }
}

clust.fdr.fmt <- clust.fdr[,c('clust','gene','logFC','FDR')]
clust.fdr.fmt$logFC <- round(clust.fdr.fmt$logFC,2)
clust.fdr.fmt$FDR <- formatC(clust.fdr.fmt$FDR, format="e", digits=2)

write.csv(clust.fdr.fmt,file='cluster_specific_pb_de_genes.csv', row.names=F)

clust.fdr %>% group_by(clust) %>% filter(abs(logFC)>0.5) %>% summarise(n())
```


Full results and cross cluster correlations
```{r}
load('cluster.specific.rda')
clust.full <- list()

for(i in 1:length(cluster.specific)){
  res <- cluster.specific[[i]]
  print(paste0('cluster ',i-1))
  flush.console()
  if(!is.null(res)){
    res.top <- topTags(res,n=nrow(res))$table
    res.top$gene <- rownames(res.top)
    res.fmt <- res.top[,c('gene','logCPM','logFC','PValue','FDR')]
    rownames(res.fmt) <- NULL
    colnames(res.fmt) <- c('Gene', 'Log Counts Per Million', 'Lead Effect Estimate (Log Fold Change)','P-Value','False Discovery Rate')
    clust.full[[paste0('clust',i-1)]] <- res.fmt
  }
}

write.xlsx(clust.full, file="Pb_DE_genes_cluster_specific.xlsx")



### correlations of estimates across clusters

exp.matrix <- lapply(clust.full,FUN=function(x) x[match(cl0$gene,x$gene),'logFC'])
exp.matrix <- data.frame(exp.matrix)

M <- cor(exp.matrix, method='spearman')
M.p <- cor.mtest(exp.matrix, conf.level=0.95)

Mp <- M
Mp[Mp==1] <- NA
Mp[lower.tri(Mp)] <- M.p$p[lower.tri(M.p$p)]

pmat <- M.p$p
pmat[upper.tri(pmat)] <- NA

colmat <- Mp
colmat[upper.tri(colmat)] <- 'black'
colmat[!upper.tri(colmat)] <- 'white'

pdf('corrplot_cluster_specific_Pb.pdf')
# corrplot.mixed(M, upper='ellipse', lower='number', lower.col='black', 
#                p.mat=M.p$p , insig='p-value', sig.level=1e-10)
#corrplot(M, method='ellipse', type='upper', addCoef.col='black')
corrplot.mixed(Mp, upper='ellipse', lower='number', addCoef.col='black',
               p.mat=pmat, sig.level=c(0.05), pch=formatC(pmat[lower.tri(pmat)],format='e',digits=1),
               insig = "label_sig", pch.col='black', pch.cex=0.6)
dev.off()
```




Volcano plots for cluster specific results
```{r}
clust.names <- c('Endothelial', 'Microglia', 'Oligodendrocytes', 'Pericytes',
                 'Astrocytes', 'Oligodendrocyte Progenitor','Choroid Plexus',
                 'Pericytes','Neurons','Microglia','Oligodendrocytes','Fibroblasts')

volcano <- list()
for(i in 1:length(cluster.specific)){
  clust.res <- cluster.specific[[i]]
  if(!is.null(clust.res)){
    res <- topTags(clust.res,n=nrow(clust.res))$table
    minFDR <- which(abs(res$FDR-0.05)==min(abs(res$FDR-0.05)))
    minFDR <- minFDR[length(minFDR)]
    pcut <- ifelse(res$FDR[1]>0.05,
                 res$PValue[1],
                 1.02*res$PValue[minFDR])
    volcano[[paste0('Clust', i-1)]] <- EnhancedVolcano(res, 
                lab=rownames(res), 
                title=paste0('Cluster ', i-1),
                subtitle=clust.names[i],
                x='logFC', 
                xlim=c(-2,2),
                y='PValue',
                pCutoff = pcut,
                FCcutoff = 0.5,
                xlab = if(i==9){bquote(~Log[2]~ "fold change")}else{""},
                ylab = if(i==4){bquote(~-Log[10]~italic(P))}else{""},
                transcriptLabSize = 4.2,
                transcriptPointSize = 1.8,
                transcriptLabhjust = 0.4,
                #drawConnectors = T,
                #boxedlabels = T,
                colAlpha = 1,
                legendPosition='top',
                caption="",
                legendVisible = F) +
      theme(plot.margin=unit(c(0,0,0,0),'cm'))
  }
}


### modifications to specific ones
res <- topTags(cluster.specific[[1]],n=nrow(cluster.specific[[1]]))$table
minFDR <- which(abs(res$FDR-0.05)==min(abs(res$FDR-0.05)))
minFDR <- minFDR[length(minFDR)]
pcut <- ifelse(res$FDR[1]>0.05,
             res$PValue[1],
             1.02*res$PValue[minFDR])
volcano[['Clust0']] <- EnhancedVolcano(res, 
                lab=rownames(res), 
                selectLab = c('Hbb-bs','Gm42726','Ermn'),
                title='Cluster 0',
                subtitle=clust.names[1],
                x='logFC', 
                xlim=c(-2,2),
                y='PValue',
                pCutoff = pcut,
                FCcutoff = 0.5,
                xlab = if(i==9){bquote(~Log[2]~ "fold change")}else{""},
                ylab = if(i==4){bquote(~-Log[10]~italic(P))}else{""},
                transcriptLabSize = 4.2,
                transcriptPointSize = 1.8,
                transcriptLabhjust = c(1.1,1.1,-0.2),
                transcriptLabvjust = 0.3,
                #drawConnectors = T,
                #boxedlabels = T,
                colAlpha = 1,
                legendPosition='top',
                caption="",
                legendVisible = F) +
      theme(plot.margin=unit(c(0,0,0,0),'cm'))

res <- topTags(cluster.specific[[2]],n=nrow(cluster.specific[[2]]))$table
minFDR <- which(abs(res$FDR-0.05)==min(abs(res$FDR-0.05)))
minFDR <- minFDR[length(minFDR)]
pcut <- ifelse(res$FDR[1]>0.05,
             res$PValue[1],
             1.02*res$PValue[minFDR])
volcano[['Clust1']] <- EnhancedVolcano(res, 
                lab=rownames(res), 
                selectLab = c('Hbb-bs','Gm42726','Kif5a','Hspa1b','Hapln2'),
                title='Cluster 1',
                subtitle=clust.names[2],
                x='logFC', 
                xlim=c(-2,2),
                y='PValue',
                pCutoff = pcut,
                FCcutoff = 0.5,
                xlab = if(i==9){bquote(~Log[2]~ "fold change")}else{""},
                ylab = if(i==4){bquote(~-Log[10]~italic(P))}else{""},
                transcriptLabSize = 4.2,
                transcriptPointSize = 1.8,
                transcriptLabhjust = -0.1,
                transcriptLabvjust = 0.3,
                drawConnectors = T,
                #boxedlabels = T,
                colAlpha = 1,
                legendPosition='top',
                caption="",
                legendVisible = F) +
      theme(plot.margin=unit(c(0,0,0,0),'cm'))

res <- topTags(cluster.specific[[5]],n=nrow(cluster.specific[[5]]))$table
minFDR <- which(abs(res$FDR-0.05)==min(abs(res$FDR-0.05)))
minFDR <- minFDR[length(minFDR)]
pcut <- ifelse(res$FDR[1]>0.05,
             res$PValue[1],
             1.02*res$PValue[minFDR])
volcano[['Clust4']] <- EnhancedVolcano(res, 
                lab=rownames(res), 
                selectLab = c('Tardbp','Extl1'),
                title='Cluster 4',
                subtitle=clust.names[5],
                x='logFC', 
                xlim=c(-2,2),
                y='PValue',
                pCutoff = pcut,
                FCcutoff = 0.5,
                xlab = if(i==9){bquote(~Log[2]~ "fold change")}else{""},
                ylab = if(i==4){bquote(~-Log[10]~italic(P))}else{""},
                transcriptLabSize = 4.2,
                transcriptPointSize = 1.8,
                #transcriptLabhjust = -0.2,
                #transcriptLabvjust = 0.3,
                drawConnectors = T,
                colConnectors = rgb(0, 0, 255, max = 255, alpha = 0),
                lengthConnectors = unit(0,'npc'),
                #boxedlabels = T,
                colAlpha = 1,
                legendPosition='top',
                caption="",
                legendVisible = F) +
      theme(plot.margin=unit(c(0,0,0,0),'cm'))

### Figure 5
tiff('Figure_5_Volcano.tiff',height=10,width=10,units='in',res=300)
  grid.arrange(grobs=volcano,
               ncol=3)
dev.off()

pdf('Figure_5_Volcano.pdf',height=10,width=10)
  grid.arrange(grobs=volcano,
               ncol=3)
dev.off()

```

Modifications of upsetR functions
```{r upsetr}
Make_main_bar <- function(Main_bar_data, Q, show_num, ratios, customQ, number_angles,
                          ebar, ylabel, ymax, scale_intersections, text_scale, attribute_plots){

  bottom_margin <- (-1)*0.65

  if(is.null(attribute_plots) == FALSE){
    bottom_margin <- (-1)*0.45
  }
  
  if(length(text_scale) > 1 && length(text_scale) <= 6){
    y_axis_title_scale <- text_scale[1]
    y_axis_tick_label_scale <- text_scale[2]
    intersection_size_number_scale <- text_scale[6]
  }
  else{
    y_axis_title_scale <- text_scale
    y_axis_tick_label_scale <- text_scale
    intersection_size_number_scale <- text_scale
  }
  
  if(is.null(Q) == F){
    inter_data <- Q
    if(nrow(inter_data) != 0){
      inter_data <- inter_data[order(inter_data$x), ]
    }
    else{inter_data <- NULL}
  }
  else{inter_data <- NULL}
  
  if(is.null(ebar) == F){
    elem_data <- ebar
    if(nrow(elem_data) != 0){
      elem_data <- elem_data[order(elem_data$x), ]
    }
    else{elem_data <- NULL}
  }
  else{elem_data <- NULL}
  
  #ten_perc creates appropriate space above highest bar so number doesnt get cut off
  if(is.null(ymax) == T){
  ten_perc <- ((max(Main_bar_data$freq)) * 0.1)
  ymax <- max(Main_bar_data$freq) + ten_perc
  }
  
  if(ylabel == "Intersection Size" && scale_intersections != "identity"){
    ylabel <- paste("Intersection Size", paste0("( ", scale_intersections, " )"))
  }
  if(scale_intersections == "log2"){
    Main_bar_data$freq <- round(log2(Main_bar_data$freq), 2)
    ymax <- log2(ymax)
  }
  if(scale_intersections == "log10"){
    Main_bar_data$freq <- round(log10(Main_bar_data$freq), 2)
    ymax <- log10(ymax)
  }
  Main_bar_plot <- (ggplot(data = Main_bar_data, aes_string(x = "x", y = "freq")) 
                    + scale_y_continuous(trans = scale_intersections)
                    + ylim(0, ymax)
                    + geom_bar(stat = "identity", width = 0.6,
                               fill = Main_bar_data$color)
                    + scale_x_continuous(limits = c(0,(nrow(Main_bar_data)+1 )), expand = c(0,0),
                                         breaks = NULL)
                    + xlab(NULL) + ylab(ylabel) +labs(title = NULL)
                    + theme(panel.background = element_rect(fill = "white"),
                            plot.margin = unit(c(0.5,0.5,bottom_margin,0.5), "lines"), panel.border = element_blank(),
                            axis.title.y = element_text(size = 8.3*y_axis_title_scale, angle=0, margin=margin(t=0,r=-130,b=0,l=0)), 
                            axis.text.y = element_text(vjust=0.3, size=7*y_axis_tick_label_scale)))
  
  #axis.title.y = element_text(hjust = 0.4, vjust=-15, size = 8.3*y_axis_title_scale, angle=90)
  
  if((show_num == "yes") || (show_num == "Yes")){
    Main_bar_plot <- (Main_bar_plot + geom_text(aes_string(label = "freq"), size = 2.2*intersection_size_number_scale, vjust = -1,
                                                angle = number_angles, colour = Main_bar_data$color))
  }
  bInterDat <- NULL
  pInterDat <- NULL
  bCustomDat <- NULL
  pCustomDat <- NULL
  bElemDat <- NULL
  pElemDat <- NULL
  if(is.null(elem_data) == F){
    bElemDat <- elem_data[which(elem_data$act == T), ]
    bElemDat <- bElemDat[order(bElemDat$x), ]
    pElemDat <- elem_data[which(elem_data$act == F), ]
  }
  if(is.null(inter_data) == F){
    bInterDat <- inter_data[which(inter_data$act == T), ]
    bInterDat <- bInterDat[order(bInterDat$x), ]
    pInterDat <- inter_data[which(inter_data$act == F), ]
  }
  if(length(customQ) != 0){
    pCustomDat <- customQ[which(customQ$act == F), ]
    bCustomDat <- customQ[which(customQ$act == T), ]
    bCustomDat <- bCustomDat[order(bCustomDat$x), ]
  }
  if(length(bInterDat) != 0){
    Main_bar_plot <- Main_bar_plot + geom_bar(data = bInterDat,
                                              aes_string(x="x", y = "freq"),
                                              fill = bInterDat$color,
                                              stat = "identity", position = "identity", width = 0.6)
  }
  if(length(bElemDat) != 0){
    Main_bar_plot <- Main_bar_plot + geom_bar(data = bElemDat,
                                              aes_string(x="x", y = "freq"),
                                              fill = bElemDat$color,
                                              stat = "identity", position = "identity", width = 0.6)
  }
  if(length(bCustomDat) != 0){
    Main_bar_plot <- (Main_bar_plot + geom_bar(data = bCustomDat, aes_string(x="x", y = "freq2"),
                                               fill = bCustomDat$color2,
                                               stat = "identity", position ="identity", width = 0.6))
  }
  if(length(pCustomDat) != 0){
    Main_bar_plot <- (Main_bar_plot + geom_point(data = pCustomDat, aes_string(x="x", y = "freq2"), colour = pCustomDat$color2,
                                                 size = 2, shape = 17, position = position_jitter(width = 0.2, height = 0.2)))
  }
  if(length(pInterDat) != 0){
    Main_bar_plot <- (Main_bar_plot + geom_point(data = pInterDat, aes_string(x="x", y = "freq"),
                                                 position = position_jitter(width = 0.2, height = 0.2),
                                                 colour = pInterDat$color, size = 2, shape = 17))
  }
  if(length(pElemDat) != 0){
    Main_bar_plot <- (Main_bar_plot + geom_point(data = pElemDat, aes_string(x="x", y = "freq"),
                                                 position = position_jitter(width = 0.2, height = 0.2),
                                                 colour = pElemDat$color, size = 2, shape = 17))
  }
  
  Main_bar_plot <- (Main_bar_plot 
                    + geom_vline(xintercept = 0, color = "gray0")
                    + geom_hline(yintercept = 0, color = "gray0"))
  
  Main_bar_plot <- ggplotGrob(Main_bar_plot)
  return(Main_bar_plot)
}

NoAttBasePlot <- function(legend, size_plot_height, Main_bar_plot, Matrix_plot, 
    hratios, Size_plot, query_legend, set_metadata, set_metadata_plots, 
    newpage) {
    top <- 1
    bottom <- 100
    if ((!is.null(legend)) && (query_legend != tolower("none"))) {
        if (query_legend == tolower("top")) {
            top <- 3
            bottom <- 102
            legend_top <- 1
            legend_bottom <- 3
            size_plot_height <- (size_plot_height + 2)
        }
        else if (query_legend == tolower("bottom")) {
            legend_top <- 101
            legend_bottom <- 103
        }
    }
    if (is.null(set_metadata)) {
        matrix_and_mainbar_right <- 100
        matrix_and_mainbar_left <- 21
        size_bar_right <- 20
        size_bar_left <- 1
    }
    else if (!is.null(set_metadata)) {
        matrix_and_mainbar_right <- set_metadata$ncols + 100
        matrix_and_mainbar_left <- set_metadata$ncols + 21
        size_bar_right <- set_metadata$ncols + 20
        size_bar_left <- set_metadata$ncols + 1
        metadata_right <- set_metadata$ncols
        metadata_left <- 1
    }
    if (newpage) {
        grid::grid.newpage()
    }
    if ((!is.null(legend)) && (query_legend != tolower("none"))) {
        if (query_legend == tolower("top")) {
            pushViewport(viewport(layout = grid.layout(102, matrix_and_mainbar_right)))
        }
        else if (query_legend == tolower("bottom")) {
            pushViewport(viewport(layout = grid.layout(103, matrix_and_mainbar_right)))
        }
    }
    else if ((is.null(legend)) || (query_legend == tolower("none"))) {
        pushViewport(viewport(layout = grid.layout(100, matrix_and_mainbar_right)))
    }
    # Modified
    vp = UpSetR:::vplayout(top:bottom, 1:(matrix_and_mainbar_right))
    pushViewport(vp)
    grid.draw(arrangeGrob(Main_bar_plot, Matrix_plot, heights = hratios))
    popViewport()
    # Modified
    # vp = UpSetR:::vplayout(size_plot_height:bottom, (matrix_and_mainbar_right-matrix_and_mainbar_left-1):96)
    # pushViewport(vp)
    # grid.draw(arrangeGrob(Size_plot))
    # popViewport()
    # if (!is.null(set_metadata)) {
    #     for (i in 1:length(set_metadata_plots)) {
    #         if (i != 1) {
    #             metadata_left <- 1 + metadata_right
    #             metadata_right <- metadata_right + set_metadata$plots[[i]]$assign
    #         }
    #         else {
    #             metadata_left <- 1
    #             metadata_right <- set_metadata$plots[[i]]$assign
    #         }
    #         vp = UpSetR:::vplayout(size_plot_height:bottom, metadata_left:metadata_right)
    #         pushViewport(vp)
    #         grid.draw(arrangeGrob(set_metadata_plots[[i]]))
    #         popViewport()
    #     }
    # }
    # if ((!is.null(legend)) && (query_legend != tolower("none"))) {
    #     vp = UpSetR:::vplayout(legend_top:legend_bottom, matrix_and_mainbar_left:matrix_and_mainbar_right)
    #     pushViewport(vp)
    #     grid.draw(arrangeGrob(legend))
    #     popViewport()
    # }
}

assignInNamespace(x="NoAttBasePlot", value=NoAttBasePlot, ns="UpSetR")
assignInNamespace(x="Make_main_bar", value=Make_main_bar, ns="UpSetR")
test()

test <- function(x){
  upset(fromList(gene.lists),sets=c('Endothelial (0)','Microglia (1)','Oligodendrocyte (2)','Pericytes (3)','Astrocytes (4)',
                                  'Oligodendrocyte Progenitor (5)','Pericytes (7)','Neurons (8)','Microglia (9)'),
             set_size.show = FALSE, mainbar.y.label = "\n\n\n\n\nNumber\nof Genes", 
      order.by='freq',keep.order=T,nintersects=35, text.scale=c(3, 2, 1.8, 2, 1.85, 1.3))
}
```

Figure 3: Bar chart, volcano, violin, and upset plot
```{r}
setwd('C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp')
load('cluster.specific.rda')
load('clustered-all.rda')


clust.top <- list()

### prep for plots
top_genes <- function(PorFDR='FDR'){
  clust.top <- list()
  for(i in 1:length(cluster.specific)){
    res <- cluster.specific[[i]]
    print(paste0('cluster ',i-1))
    flush.console()
    if(!is.null(res)){
      res.top <- topTags(res,n=nrow(res))$table
      if(PorFDR=='P'){
        res.top <- res.top[res.top$PValue<0.05,]
      }else if(PorFDR=='FDR')
      {
        res.top <- res.top[res.top$FDR<0.05,]
      }
      if(nrow(res.top)!=0){
        print(paste0('has nominal genes'))
        flush.console()
        res.top$gene <- rownames(res.top)
        res.top$clust <- i-1
        rownames(res.top) <- NULL
        clust.top[[paste0('cluster ',i-1)]] <- res.top
      }
    }
  }
  clust.top
}

#prep for upset plot
clust.top <- top_genes(PorFDR='P')
gene.lists <- lapply(clust.top, FUN=function(x) x$gene)

#prep for bar chart
updown <- data.frame(
  reg = c(rep('Up',9),rep('Down',9)),
  clust = as.character(rep(c(0,1,2,3,4,5,7,8,9),2)),
  ngene = c(sapply(clust.top,FUN=function(x) table(x$logFC>0)['TRUE']),
            -1*sapply(clust.top,FUN=function(x) table(x$logFC>0)['FALSE']))
)

#prep for volcano/violin plots 
clust1 <- subset(All, subset = seurat_clusters==1)

clust1.res <- cluster.specific[[2]]
clust1.res <- topTags(clust1.res,n=nrow(clust1.res))$table

clust0.res <- cluster.specific[[1]]
clust0.res <- topTags(clust0.res,n=nrow(clust0.res))$table

clust0.res <- clust0.res[match(rownames(clust1.res),rownames(clust0.res)),]



text.size <- 14
text.size.leg <- 18
point.size <- 0.2

#tSNE by Pb
f3a <- DimPlot(object = All,pt.size=point.size,reduction='tsne',group.by='Treat') + 
  theme(text=element_text(size=text.size), legend.text=element_text(size=text.size.leg))

#bar chart up/down regulated genes
f3b <- ggplot(updown, aes(x=clust,y=ngene,fill=reg)) +
  geom_bar(stat='identity',position='identity') +
  labs(fill='Expression with Lead',
       x='Cell Cluster',
       y='N Genes with p-value < 0.05') + 
  guides(fill= guide_legend(reverse=T)) + theme_bw() + 
  theme(legend.position='top', axis.text=element_text(size=text.size), axis.title=element_text(size=text.size), 
        legend.text=element_text(size=text.size.leg), legend.title=element_text(size=text.size.leg)) +
  scale_fill_manual(values=c("firebrick3","royalblue4"))

#volcano cluster 1
# f3c <- EnhancedVolcano(clust1.res, 
#                 lab=rownames(clust1.res), 
#                 selectLab=c('Hbb-bs','Gm42726','Kif5a'),
#                 title='',
#                 subtitle='',
#                 x='logFC', 
#                 xlim=c(-1,1),
#                 y='FDR',
#                 pCutoff = 0.05,
#                 FCcutoff = 0.5,
#                 ylab = bquote(~-Log[10]~adjusted~italic(P)),
#                 transcriptLabSize = 4.5,
#                 transcriptPointSize = 1.4,
#                 transcriptLabvjust=0.1,
#                 transcriptLabhjust=-0.1,
#                 #drawConnectors = T,
#                 #boxedlabels = T,
#                 colAlpha = 1,
#                 legendVisible = F,
#                 caption="") +
#   theme(plot.margin = margin(-10,4,0,2))

#vln Gm42726
f3d <- VlnPlot(All, features = "Gm42726", point.size.use=point.size, group.by = 'Treat') + 
  geom_boxplot(alpha=0.7, outlier.shape=NA, show.legend=F) +
  theme(text=element_text(size=text.size), legend.position="none", 
        axis.title.x=element_blank(), axis.text=element_text(size=text.size))
#vln Hbb-bs
f3e <- VlnPlot(All, features = "Hbb-bs", point.size.use=point.size, group.by = 'Treat') + 
  geom_boxplot(alpha=0.7, outlier.shape=NA, show.legend=F) +
  theme(text=element_text(size=text.size), legend.position="none", 
        axis.title.x=element_blank(), axis.text=element_text(size=text.size))

#upset of nominal genes
names(gene.lists) <- c('Endothelial (0)','Microglia (1)','Oligodendrocyte (2)','Pericytes (3)','Astrocytes (4)',
                                  'Oligodendrocyte Progenitor (5)','Pericytes (7)','Neurons (8)','Microglia (9)')
f3g <- upset(fromList(gene.lists),sets=c('Endothelial (0)','Microglia (1)','Oligodendrocyte (2)','Pericytes (3)','Astrocytes (4)',
                                  'Oligodendrocyte Progenitor (5)','Pericytes (7)','Neurons (8)','Microglia (9)'),
             set_size.show = FALSE, mainbar.y.label = "\n\n\n\n\n\n\nNumber\nof Genes", 
      order.by='freq',keep.order=T,nintersects=35, text.scale=c(2.3, 2, 1.8, 2, 1.4, 1.3)) 
      #c(intersection size title, intersection size tick labels, set size title, set size tick labels, set names, numbers above bars)

# f3g <- upset(fromList(gene.lists),sets=c('cluster 0','cluster 1','cluster 2','cluster 3','cluster 4',
#                                   'cluster 5','cluster 7','cluster 8','cluster 9'),
#              set_size.show = FALSE, mainbar.y.label = "Number of Genes", 
#       order.by='freq',keep.order=T,nintersects=35, text.scale=c(3, 2, 1.8, 2, 1.65, 1.3)) 

f3g #hackish way to get upset to play along
grid.edit('arrange',name='arrange2')
f3g_grob <- grid.grab()

#scatter clust 0 vs clust 1
# f3f <- {~smoothScatter(clust0.res$logFC,clust1.res$logFC,
#               xlab='Cluster 0 logFC estimates',
#               ylab='Cluster 1 logFC estimates')
# ~abline(h=0,v=0)
# ~text(-0.7,0.7,paste0('r=',round(cor(clust0.res$logFC,clust1.res$logFC,method='spearman'),2)),cex=1.3)}
# grid.edit('arrange',name='arrange2')
# f3f <- grid.grab()
f3f.df <- data.frame('cl0'=clust0.res$logFC,'cl1'=clust1.res$logFC, 
                     d= densCols(clust0.res$logFC,clust1.res$logFC, colramp=colorRampPalette(blues9[-(1:3)])))
f3f <- ggplot(f3f.df,aes(x=cl0, y=cl1, col=d)) +
  geom_hline(yintercept = 0, col='darkgrey') + geom_vline(xintercept = 0, col='darkgrey') +
  geom_point(size=0.8) +
  scale_color_identity() +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        axis.text=element_text(size=text.size), axis.title=element_text(size=text.size)) +
  xlab("Lead Effect on \n Eplithelial Gene Expression") + ylab("Lead Effect on \n Microglia Gene Expression") +
  annotate("text", x=-0.6, y=0.7, label=paste0('r=',round(cor(clust0.res$logFC,clust1.res$logFC,method='spearman'),2)), size=text.size-8)
  

fig3_plots <- list(f3a, f3b, f3d, f3e, f3f, f3g_grob)
fig3_label <- list("A.", "B.", "C.", "D.", "E.", "F.")
fig3_plots <- lapply(1:length(fig3_plots), FUN = function(i){
  arrangeGrob(fig3_plots[[i]], top = textGrob(fig3_label[[i]], x= unit(0, "npc")
                                ,y= unit(1, "npc"), just= c("left","top"),
                                gp=gpar(col="black", fontsize=18)))
})

tiff("Figure_3_DE_clust.tif", height=14, width=10, units="in", res=300)
grid.arrange(fig3_plots[[1]], fig3_plots[[2]], fig3_plots[[3]], fig3_plots[[4]], fig3_plots[[5]], fig3_plots[[6]],
             heights = c(1,1,1.6),
             layout_matrix=rbind(c(1,1,1,2,2,2),
                                 c(3,3,4,4,5,5),
                                 c(7,7,7,7,7,7)))
dev.off()



pdf("Figure_3_DE_clust.pdf", height=14, width=10)
grid.arrange(fig3_plots[[1]], fig3_plots[[2]], fig3_plots[[3]], fig3_plots[[4]], fig3_plots[[5]], fig3_plots[[6]],
             heights = c(1,1,1.6),
             layout_matrix=rbind(c(1,1,1,2,2,2),
                                 c(3,3,4,4,5,5),
                                 c(7,7,7,7,7,7)))
dev.off()



```


Check if top bulk results are cluster markers 
```{r}
#load bulk results
clust.bulk.res <- readRDS('Pb.clustALL.fit.rds')
clust.bulk.res <- topTags(clust.bulk.res,n=nrow(clust.bulk.res))$table

#load cluster markers

path <- 'C:/Users/John/Google Drive/Pb_Hipp_Research/Single Cell Hippo Manuscript/Cluster markers padj05.xlsx'
clust.markers <- path %>%
  excel_sheets() %>%
  purrr::set_names() %>%
  purrr::map(read_excel, path=path)

head(clust.bulk.res)
table(clust.bulk.res$FDR<0.05)
head(clust.markers[[1]])

marker.overlap <- lapply(clust.markers, FUN=function(markers){
  markers <- markers[1:10,]
  intersect(markers$gene, rownames(clust.bulk.res[clust.bulk.res$FDR<0.05,]))
})

marker.overlap

clust.bulk.res$clust0 <- rownames(clust.bulk.res) %in% clust.markers[[1]]$gene[1:10] 
clust.bulk.res$clust1 <- rownames(clust.bulk.res) %in% clust.markers[[2]]$gene[1:10] 
clust.bulk.res$clust2 <- rownames(clust.bulk.res) %in% clust.markers[[3]]$gene[1:10] 
clust.bulk.res$clust5 <- rownames(clust.bulk.res) %in% clust.markers[[6]]$gene[1:10] 
clust.bulk.res$clust10 <- rownames(clust.bulk.res) %in% clust.markers[[11]]$gene[1:10] 

table(clust.bulk.res$FDR<0.05,clust.bulk.res$clust0)
fisher.test(clust.bulk.res$FDR<0.05,
            clust.bulk.res$clust0)

table(clust.bulk.res$FDR<0.05,clust.bulk.res$clust1)
fisher.test(clust.bulk.res$FDR<0.05,
            clust.bulk.res$clust1)

table(clust.bulk.res$FDR<0.05,clust.bulk.res$clust2)
fisher.test(clust.bulk.res$FDR<0.05,
            clust.bulk.res$clust2)

table(clust.bulk.res$FDR<0.05,clust.bulk.res$clust5)
fisher.test(clust.bulk.res$FDR<0.05,
            clust.bulk.res$clust5)

table(clust.bulk.res$FDR<0.05,clust.bulk.res$clust10)
fisher.test(clust.bulk.res$FDR<0.05,
            clust.bulk.res$clust10)

fisher.overlap <- lapply(clust.markers, FUN=function(markers){
  #markers <- markers[1:10,]
  is.marker <- rownames(clust.bulk.res) %in% markers$gene
  fisher.test(clust.bulk.res$FDR<0.05, is.marker)
})

table.overlap <- lapply(clust.markers, FUN=function(markers){
  #markers <- markers[1:10,]
  is.marker <- rownames(clust.bulk.res) %in% markers$gene
  table(clust.bulk.res$FDR<0.05, is.marker)
})

```



Preparing results for LRpath
```{r lrpath}


tab.path <- "C:/Users/John/Google Drive/Pb_Hipp_Research/Results"
seurat.path <- "C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp"

#prepare LR path file for overall results
res.all <- read.csv(file.path(tab.path,"Pb_DE_genes_across_all_clusters.csv"),header=T)
names(res.all)[1] <- "geneid"
res.all <- res.all[,c(1,5,2,3)]
entrez <- mapIds(org.Mm.eg.db, as.character(res.all$geneid), "ENTREZID", "SYMBOL")
res.all$geneid <- entrez
res.all <- res.all[!is.na(res.all$geneid),]
write.table(res.all, file=file.path(seurat.path,"LR_all_test2.txt"),row.names=F,quote=F,sep="\t",na="")


### write tables for cluster specific results
load(file.path(seurat.path,"cluster.specific.rda"))


lapply(names(clust.full), FUN=function(clust){
  res <- clust.full[[clust]]
  names(res)[2] <- "geneid"
  res <- res[,c(2,5,4,3)]
  entrez <- mapIds(org.Mm.eg.db, as.character(res$geneid), "ENTREZID", "SYMBOL")
  res$geneid <- entrez
  res <- res[!is.na(res$geneid),]
  write.table(res, file=paste0(seurat.path,"/LR_",clust,".txt"),row.names=F,quote=F,sep="\t",na="")
})


#prepare LR path file for genes in at least two clusters
clust.nom <- lapply(clust.full,FUN=function(x){x[x$PValue<0.05,'gene']})
nom.all <- unlist(clust.nom)
nom.shared <- table(nom.all)
count.shared <- names(nom.shared)[nom.shared >= 2]

all.genes <- factor(as.integer(clust.full[[1]]$gene %in% count.shared))
names(all.genes) <- clust.full[[1]]$gene


```


After LR path
```{r lr_after}
LR.path <- "C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp/LRpath"


LR.all <- read.xlsx(file.path(LR.path,"all_logcpm.xlsx"))
LR.all <- LR.all[order(LR.all$FDR),]
LR.all <- LR.all[LR.all$ConceptType=="GOBP",]

write.xlsx(LR.all, file=file.path(tab.path,"LR_path_overall.xlsx"),row.names=F,quote=F)


LR.clust <- list()
clusts <- paste0("clust", c(0,1,2,3,4,5,7,8,9))

for(file in clusts){
  LR.clust[[file]] <- read.xlsx(file.path(LR.path,paste0(file, "_LR.xlsx")))
}

write.xlsx(LR.clust, file=file.path(tab.path,"LR_path_clust_specific.xlsx"))
```


Figure 6
```{r fig6}

setwd("C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp")
LR.path <- "C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp/LRpath"

LR.clust <- list()
clusts <- paste0("clust", c(0,1,2,3,4,5,7,8,9))

for(file in clusts){
  LR.clust[[file]] <- read.xlsx(file.path(LR.path,paste0(file, "_LR.xlsx")))
}

#upset plot set up
top.paths <- lapply(LR.clust, FUN=function(x){
  x <- x[x$FDR<0.05,]
  x$Id
})


n.fdr <- sapply(top.paths, length)
barplot(n.fdr, main='Number of FDR significant pathways', las=2)
top.paths <- top.paths[which(names(top.paths)!="clust4")]
names(top.paths) <- c('Endothelial (0)','Microglia (1)','Oligodendrocytes (2)','Pericytes (3)','Oligodendrocyte Progenitor (5)','Pericytes (7)','Neurons (8)','Microglia (9)')

#barplots set up
load('cluster.specific.rda')
clust.full <- list()

for(i in 1:length(cluster.specific)){
  res <- cluster.specific[[i]]
  print(paste0('cluster ',i-1))
  flush.console()
  if(!is.null(res)){
    res.top <- topTags(res,n=nrow(res))$table
    res.top$gene <- rownames(res.top)
    res.top$clust <- i-1
    res.fmt <- res.top[,c('clust','gene','logCPM','logFC','PValue','FDR')]
    rownames(res.fmt) <- NULL
    clust.full[[paste0('clust',i-1)]] <- res.fmt
  }
}

prot.fold.genes <- genesInTerm(clust.GO[[1]], 'GO:0006457')[[1]]
prot.fold.fc <- clust.full[[2]]
prot.fold.fc <- prot.fold.fc[prot.fold.fc$gene %in% prot.fold.genes,]
prot.fold.fc <- prot.fold.fc[order(-prot.fold.fc$logFC),]
prot.fold.fc$gene <- factor(prot.fold.fc$gene, levels=prot.fold.fc$gene)
prot.fold.fc$col <- ifelse(prot.fold.fc$PValue<0.05,'blue','red')

int12.genes <- genesInTerm(clust.GO[[1]], 'GO:0006950')[[1]]
int12.fc <- clust.full[[3]]
int12.fc <- int12.fc[int12.fc$gene %in% int12.genes,]
int12.fc <- int12.fc[order(-int12.fc$logFC),]
int12.fc$gene <- factor(int12.fc$gene, levels=int12.fc$gene)
int12.fc$col <- ifelse(int12.fc$PValue<0.05,'blue','red')


DNArep.genes <- genesInTerm(clust.GO[[1]], 'GO:0006303')[[1]]
DNArep.fc <- clust.full[[7]]
DNArep.fc <- DNArep.fc[DNArep.fc$gene %in% DNArep.genes,]
DNArep.fc <- DNArep.fc[order(-DNArep.fc$logFC),]
DNArep.fc$gene <- factor(DNArep.fc$gene, levels=DNArep.fc$gene)
DNArep.fc$col <- ifelse(DNArep.fc$PValue<0.05,'blue','red')



text.size=14

#upset plot of FDR significant pathways
f6a <- upset(fromList(top.paths), sets=names(top.paths),
             mainbar.y.label='\n\n\n\n\n\nNumber of\nGene Set\nPathways',
      order.by='freq',keep.order=T,nintersects = NA, text.scale=c(2, 1.6, 1.5, 1.7, 1.4, 1.2)) #c(intersection size title, intersection size tick labels, set size title, set size tick labels, set names, numbers above bars)
f6a
grid.edit('arrange',name='arrange2')
f6a_grob <- grid.grab()


#plot genes in pathway of interest

f6b <- ggplot(prot.fold.fc, aes(x=gene, y=logFC, fill=col)) +
  geom_bar(stat="identity", width=1) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank(),
        legend.position = "none", axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        axis.line.y = element_line(), text=element_text(size=text.size), plot.title = element_text(size=text.size-2)) +
  xlab("") + ylab("LogFC with Lead") + ggtitle("Cluster 1 (Microglia)\nProtein Folding Pathway") +
  scale_fill_manual(values=c("blue","red"))

f6c <- ggplot(int12.fc, aes(x=gene, y=logFC, fill=col)) +
  geom_bar(stat="identity", width=1) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank(),
        legend.position = "none", axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        axis.line.y = element_line(), text=element_text(size=text.size), plot.title = element_text(size=text.size-2)) +
  xlab("") + ylab("LogFC with Lead") + ggtitle("Cluster 2 (Oligodendrocytes)\nInterleukin-12 Biosynthetic Process") +
  scale_fill_manual(values=c("blue","red"))

f6d <- ggplot(DNArep.fc, aes(x=gene, y=logFC, fill=col)) +
  geom_bar(stat="identity", width=1) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.border = element_blank(),
        legend.position = "none", axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        axis.line.y = element_line(), text=element_text(size=text.size), plot.title = element_text(size=text.size-2)) +
  xlab("") + ylab("LogFC with Lead") + ggtitle("Cluster 7 (Pericytes)\nDouble-Strand Break Repair") +
  scale_fill_manual(values=c("blue","red"))

fig6_plots <- list(f6a_grob, f6b, f6c, f6d)
fig6_label <- list("A.", "B.", "C.", "D.")
fig6_plots <- lapply(1:length(fig6_plots), FUN = function(i){
  arrangeGrob(fig6_plots[[i]], top = textGrob(fig6_label[[i]], x= unit(0, "npc")
                                ,y= unit(1, "npc"), just= c("left","top"),
                                gp=gpar(col="black", fontsize=18)))
})

dummy <- ggplot(DNArep.fc, aes(x=gene, y=logFC, fill=col)) +
  geom_bar(stat="identity", width=1) +
  theme() + scale_fill_manual(values=c("blue","red"), labels=c(expression(P < 0.05), expression(P >= 0.05))) +
  labs(fill="")
legend <- get_legend(dummy)


tiff("Figure_6_Path.tif", height=8, width=12, units="in", res=300)
grid.arrange(fig6_plots[[1]], fig6_plots[[2]], fig6_plots[[3]], fig6_plots[[4]], legend,
             heights = c(2,1.1), widths=c(1,1,1,0.3),
             layout_matrix=rbind(c(1,1,1,1),
                                 c(2,3,4,5)))
dev.off()



pdf("Figure_6_Path.pdf", height=8, width=12)
grid.arrange(fig6_plots[[1]], fig6_plots[[2]], fig6_plots[[3]], fig6_plots[[4]], legend,
             heights = c(2,1.1), widths=c(1,1,1,0.3),
             layout_matrix=rbind(c(1,1,1,1),
                                 c(2,3,4,5)))
dev.off()




```

# MuSiC
```{r music}
library(MuSiC)
library(BisqueRNA)
library(Seurat)
library(Biobase)

### bulk human progenitor neuron data - prepare in ExpressionSet
#Jiang P. 2017. RNA-Seq of Human Neural Progenitor Cells Exposed to Lead (Pb) Reveals Transcriptome Dynamics, Splicing Alterations and Disease Risk Associations 
hmn.prog <- read.table("C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp/HumanNeuroProg/ExpectCounts_all_samples.txt", header=T)
hmn.prog.matrix <- as.matrix(hmn.prog[,-c(1,2)])
rownames(hmn.prog.matrix) <- hmn.prog$Gene.ID

hmn.prog.pd <- data.frame(treat=gsub('_.*','',colnames(hmn.prog.matrix)),
                          time=gsub('.*_','',colnames(hmn.prog.matrix)))
rownames(hmn.prog.pd) <- colnames(hmn.prog.matrix)
hmn.prog.pd <- AnnotatedDataFrame(hmn.prog.pd)

hmn.prog.eset <- ExpressionSet(assayData=hmn.prog.matrix, phenoData=hmn.prog.pd)


#### denate gyrus neurogenesis dataset
#Hochgerner, H., Zeisel, A., Lönnerberg, P. et al. Conserved properties of dentate gyrus neurogenesis across postnatal development revealed by single-cell RNA sequencing. Nat Neurosci 21, 290–299 (2018). 
neu.ref <- read.table("C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp/HumanNeuroProg/GSE95315_10X_expression_data_v2.tab", header=T)
neu.ref <- t(neu.ref)
colnames(neu.ref) <- neu.ref[1,]
neu.ref <- neu.ref[-1,]
clusters <- as.character(neu.ref[,2])

neu.ref.matrix <- as.matrix(neu.ref[,-c(1,2)])
storage.mode(neu.ref.matrix) <- "numeric"
neu.ref.matrix <- t(neu.ref.matrix)

neu.ref.pd <- data.frame(pool=gsub('_.*','',rownames(neu.ref)),
                         age=neu.ref[,1],
                         cluster=clusters)
identical(rownames(neu.ref.pd), colnames(neu.ref.matrix))
neu.ref.pd <- AnnotatedDataFrame(neu.ref.pd)

neu.ref.eset <- ExpressionSet(assayData=neu.ref.matrix, phenoData=neu.ref.pd)

library(biomaRt)
human = useMart("ensembl", dataset = "hsapiens_gene_ensembl")
mouse = useMart("ensembl", dataset = "mmusculus_gene_ensembl")

genes.human = getLDS(attributes = c("mgi_symbol"), filters = "mgi_symbol", values = rownames(neu.ref.eset) , mart = mouse, attributesL = c("hgnc_symbol"), martL = human, uniqueRows=T)
table(rownames(neu.ref.eset) %in% genes.human$MGI.symbol)

#for now, dumping genes that are non-uniquely matching that don't have a gene symbol that is same (i.e. ones not like Rln1 and RLN1)
library(dplyr)
genes.human.uni <- genes.human %>% group_by(MGI.symbol) %>% filter(n()==1)
genes.human.nonuni <- genes.human %>% group_by(MGI.symbol) %>% filter(n()>1)
genes.human.unipick <- genes.human.nonuni %>% group_by(MGI.symbol) %>% filter(toupper(MGI.symbol)==HGNC.symbol)
table(unique(genes.human.nonuni$MGI.symbol) %in% genes.human.unipick$MGI.symbol)

mouse.to.human <- rbind(genes.human.unipick, genes.human.uni) %>% data.frame
m2h <- mouse.to.human$HGNC.symbol
names(m2h) <- mouse.to.human$MGI.symbol
m2h <- m2h[!duplicated(m2h)] #have to get rid of non-unique human for rownames to work....
m2h <- m2h[m2h!='PISD'] #AHHH there is a Pisd and PISD in original, remove Psid to stop dup rowname

rownames(neu.ref.eset) <- ifelse(rownames(neu.ref.eset) %in% names(m2h), m2h[rownames(neu.ref.eset)], rownames(neu.ref.eset))


#### single cell mouse hipp data

# load('C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp/clustered-all.rda')
# 
# ms.hipp.eset <- SeuratToExpressionSet(All, delimiter='_', position=1, version='v3')
# 
# library(biomaRt)
# human = useMart("ensembl", dataset = "hsapiens_gene_ensembl")
# mouse = useMart("ensembl", dataset = "mmusculus_gene_ensembl")
#  
# genes.human = getLDS(attributes = c("mgi_symbol"), filters = "mgi_symbol", values = rownames(ms.hipp.eset) , mart = mouse, attributesL = c("hgnc_symbol"), martL = human, uniqueRows=T)
# table(rownames(ms.hipp.eset) %in% genes.human$MGI.symbol)
# 
# #for now, dumping genes that are non-uniquely matching that don't have a gene symbol that is same (i.e. ones not like Rln1 and RLN1)
# library(dplyr)
# genes.human.uni <- genes.human %>% group_by(MGI.symbol) %>% filter(n()==1)
# genes.human.nonuni <- genes.human %>% group_by(MGI.symbol) %>% filter(n()>1)
# genes.human.unipick <- genes.human.nonuni %>% group_by(MGI.symbol) %>% filter(toupper(MGI.symbol)==HGNC.symbol) 
# table(unique(genes.human.nonuni$MGI.symbol) %in% genes.human.unipick$MGI.symbol) 
# 
# mouse.to.human <- rbind(genes.human.unipick, genes.human.uni) %>% data.frame
# m2h <- mouse.to.human$HGNC.symbol
# names(m2h) <- mouse.to.human$MGI.symbol 
# m2h <- m2h[!duplicated(m2h)] #have to get rid of non-unique human for rownames to work....
# m2h <- m2h[m2h!='PISD'] #AHHH there is a Pisd and PISD in original, remove Psid to stop dup rowname
# 
# rownames(ms.hipp.eset) <- ifelse(rownames(ms.hipp.eset) %in% names(m2h), m2h[rownames(ms.hipp.eset)], rownames(ms.hipp.eset))
# 
# #name clusters more informative
# clust.names <- c('endothelial','microglia','oligodendrocytes','pericytes','astrocytes','oligo progenitor','choroid plexus','pericytes','neurons','microglia','oligodendrocytes','fibroblasts')
# names(clust.names) <- c('0','1','2','3','4','5','6','7','8','9','10','11')
# table(ms.hipp.eset$cellType)
# table(clust.names[ms.hipp.eset$cellType])
# ms.hipp.eset$cellType <- clust.names[ms.hipp.eset$cellType]

#estimate proportions
library(xbioc)

# cell.est <- music_prop(bulk.eset= hmn.prog.eset, sc.eset=ms.hipp.eset,
#                        clusters='cellType', samples='SubjectName',
#                        #select.ct=NULL)
#                        select.ct=c('oligodendrocytes','pericytes','astrocytes','neurons','oligo progenitor'))
cell.est <- music_prop(bulk.eset= hmn.prog.eset, sc.eset=neu.ref.eset,
                       clusters='cluster', samples='pool',
                       select.ct=NULL)
                       #select.ct=c('oligodendrocytes','pericytes','astrocytes','neurons','oligo progenitor'))

head(cell.est$Est.prop.weighted)


test <- cell.est$Est.prop.weighted %>% data.frame
test$time <- gsub('.*_D','',rownames(test)) %>% as.numeric
test$treat <- gsub('_.*','',rownames(test))

library(ggplot2)

pdf('C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp/MuSiC predictions 2.pdf')

  ggplot(data=test, aes(x=time, y=NFOL, group=treat)) +
    geom_line(aes(col=treat)) + ylim(c(0,1))

  ggplot(data=test, aes(x=time, y=OPC, group=treat)) +
  geom_line(aes(col=treat)) + ylim(c(0,1))
  
  ggplot(data=test, aes(x=time, y=nIPC, group=treat)) +
  geom_line(aes(col=treat)) + ylim(c(0,1))
  
  ggplot(data=test, aes(x=time, y=Neuroblast_1, group=treat)) +
  geom_line(aes(col=treat)) + ylim(c(0,1))
  
  ggplot(data=test, aes(x=time, y=Pericytes, group=treat)) +
  geom_line(aes(col=treat)) + ylim(c(0,1))
  
  ggplot(data=test, aes(x=time, y=Mossy.Adcyap1, group=treat)) +
  geom_line(aes(col=treat)) + ylim(c(0,1))
  
  ggplot(data=test, aes(x=time, y=Mossy.Cyp26b1, group=treat)) +
  geom_line(aes(col=treat)) + ylim(c(0,1))
  
  ggplot(data=test, aes(x=time, y=GABA.Cnr1, group=treat)) +
  geom_line(aes(col=treat)) + ylim(c(0,1))

  ggplot(data=test, aes(x=time, y=GABA.Lhx6, group=treat)) +
  geom_line(aes(col=treat)) + ylim(c(0,1))
  
  ggplot(data=test, aes(x=time, y=Cajal.Retzius, group=treat)) +
  geom_line(aes(col=treat)) + ylim(c(0,1))

dev.off()



p.opc <- ggplot(data=test, aes(x=time, y=OPC, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Oligodendrocyte\nPrecursor') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.nipc <- ggplot(data=test, aes(x=time, y=nIPC, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Neuronal\nIntermediate Progenitor') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.neub <- ggplot(data=test, aes(x=time, y=Neuroblast_1, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Neuroblast') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.dummy <- ggplot(data=test, aes(x=time, y=Neuroblast_1, group=treat)) +
  geom_line(aes(col=treat)) + labs(color='Treatment') + scale_color_discrete(labels=c('Control', 'Lead 5 μM', 'Lead 30 μM')) +
  theme(legend.text=element_text(size=18), legend.title=element_text(size=18))
p.predleg <- get_legend(p.dummy)

# pb hipp as reference
# pdf('C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp/MuSiC predictions.pdf')
#   ggplot(data=test, aes(x=time, y=neurons, group=treat)) +
#   geom_line(aes(col=treat)) + ylim(c(0,1))
# 
#   ggplot(data=test, aes(x=time, y=oligo.progenitor, group=treat)) +
#   geom_line(aes(col=treat)) + ylim(c(0,1))
#   
#   ggplot(data=test, aes(x=time, y=pericytes, group=treat)) +
#   geom_line(aes(col=treat)) + ylim(c(0,1))
#   
#   ggplot(data=test, aes(x=time, y=astrocytes, group=treat)) +
#   geom_line(aes(col=treat)) + ylim(c(0,1))
# dev.off()




### specific marker time course

# using the TPM version for looking at individual genes
gene.time <- hmn.prog.pd@data
hmn.prog.matrix <- read.table("C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp/HumanNeuroProg/TPM_all_samples.txt", header=T)
rownames(hmn.prog.matrix) <- hmn.prog.matrix[,1]
hmn.prog.matrix <- hmn.prog.matrix[,-c(1,2)]
hmn.prog.matrix <- as.matrix(hmn.prog.matrix)

gene.time$time <- gsub('D','',gene.time$time) %>% as.numeric()
gene.time$OLIG1 <- hmn.prog.matrix['OLIG1',rownames(gene.time)]
gene.time$OLIG2 <- hmn.prog.matrix['OLIG2',rownames(gene.time)]
gene.time$OLIG3 <- hmn.prog.matrix['OLIG3',rownames(gene.time)]
gene.time$PDGFRA <- hmn.prog.matrix['PDGFRA',rownames(gene.time)]
gene.time$MBP <- hmn.prog.matrix['MBP',rownames(gene.time)]

gene.time$HAPLN2 <- hmn.prog.matrix['HAPLN2',rownames(gene.time)]
gene.time$HBB <- hmn.prog.matrix['HBB',rownames(gene.time)]
gene.time$KIF5A <- hmn.prog.matrix['KIF5A',rownames(gene.time)]
gene.time$PTGDS <- hmn.prog.matrix['PTGDS',rownames(gene.time)]
gene.time$CRYAB <- hmn.prog.matrix['CRYAB',rownames(gene.time)]

ggplot(data=gene.time, aes(x=time, y=KIF5A, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') +
  theme(axis.title=element_text(size=14), axis.text=element_text(size=12))


p.olig1 <- ggplot(data=gene.time, aes(x=time, y=OLIG1, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') +
  ylab('OLIG1') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.olig2 <- ggplot(data=gene.time, aes(x=time, y=OLIG2, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') +
  ylab('OLIG2') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

# p.olig3 <- ggplot(data=gene.time, aes(x=time, y=OLIG3, group=treat)) +
#   geom_line(aes(col=treat), size=1.3) +
#   theme_bw() + xlab('Days') +
#   ylab('OLIG3') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.pdgfra <- ggplot(data=gene.time, aes(x=time, y=PDGFRA, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') +
  ylab('PDGFRA') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.mbp <- ggplot(data=gene.time, aes(x=time, y=MBP, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') +
  ylab('MBP') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))


fig_pred_plots <- list(p.opc, p.nipc, p.neub, p.olig1, p.olig2, p.pdgfra, p.mbp)
fig_pred_label <- list("A.", "B.", "C.", "D.", "E.", "F.", "G.")
fig_pred_plots <- lapply(1:length(fig_pred_plots), FUN = function(i){
  arrangeGrob(fig_pred_plots[[i]], top = textGrob(fig_pred_label[[i]], x= unit(0, "npc")
                                ,y= unit(1, "npc"), just= c("left","top"),
                                gp=gpar(col="black", fontsize=18)))
})



### Figure 4

tiff("C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp/Figure_4_MuSiC.tif", height=8, width=12, units="in", res=300)
grid.arrange(fig_pred_plots[[1]], fig_pred_plots[[2]], fig_pred_plots[[3]],
             p.predleg, fig_pred_plots[[4]], fig_pred_plots[[5]], 
                        fig_pred_plots[[6]], fig_pred_plots[[7]],
             heights = c(1.5,1,1),
             widths = c(1/2, 1/3, 2/3, 1/6, 5/6),
             layout_matrix=rbind(c(1,1,2,2,3),
                                 c(4,5,5,6,6),
                                 c(4,7,7,8,8)))
dev.off()

pdf("C:/Users/John/Google Drive/Misc/RNAseq/Pb_Hipp/Figure_4_MuSiC.pdf", height=8, width=12)
grid.arrange(fig_pred_plots[[1]], fig_pred_plots[[2]], fig_pred_plots[[3]],
             p.predleg, fig_pred_plots[[4]], fig_pred_plots[[5]], 
                        fig_pred_plots[[6]], fig_pred_plots[[7]],
             heights = c(1.5,1,1),
             widths = c(1/2, 1/3, 2/3, 1/6, 5/6),
             layout_matrix=rbind(c(1,1,2,2,3),
                                 c(4,5,5,6,6),
                                 c(4,7,7,8,8)))
dev.off()



### supplementary other cell prop predictions

p.end <- ggplot(data=test, aes(x=time, y=Endothelial, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('\nEndothelial') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.vlcm <- ggplot(data=test, aes(x=time, y=VLMC, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Vascular and\nLeptomeningeal') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.mcr <- ggplot(data=test, aes(x=time, y=Microglia, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('\nMicroglia') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.pvm <- ggplot(data=test, aes(x=time, y=PVM, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Perivascular\nMacrophage') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.ol <- ggplot(data=test, aes(x=time, y=OL, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('\nOligodendrocyte') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.ast <- ggplot(data=test, aes(x=time, y=Astrocytes, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('\nAstrocytes') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.rgl <- ggplot(data=test, aes(x=time, y=Radial_Glia.like, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('\nRadial Glia-like') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.neub2 <- ggplot(data=test, aes(x=time, y=Neuroblast_2, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('\nNeuroblast_2') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.grni <- ggplot(data=test, aes(x=time, y=Granule.immature, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Neuron\n(Granule-immature)') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.grnm <- ggplot(data=test, aes(x=time, y=Granule.mature, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Neuron\n(Granule-mature)') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.mcyp <- ggplot(data=test, aes(x=time, y=Mossy.Cyp26b1, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Neuron\n(Mossy-Cyp26b1)') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.madc <- ggplot(data=test, aes(x=time, y=Mossy.Adcyap1, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Neuron\n(Mossy-Adcyap1)') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.mklk <- ggplot(data=test, aes(x=time, y=Mossy.Klk8, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Neuron\n(Mossy-Klk8)') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.gabac <- ggplot(data=test, aes(x=time, y=GABA.Cnr1, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Neuron\n(GABA-Cnr1)') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.gabal <- ggplot(data=test, aes(x=time, y=GABA.Lhx6, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Neuron\n(GABA-Lhx6)') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

p.cr <- ggplot(data=test, aes(x=time, y=Cajal.Retzius, group=treat)) +
  geom_line(aes(col=treat), size=1.3) +
  theme_bw() + xlab('Days') + ylim(c(0,1)) +
  ylab('Neuron\n(Cajal-Retzius)') + theme(legend.position = 'none', axis.title=element_text(size=14), axis.text=element_text(size=12))

tiff('MuSiC other cell types.tif', width=12, height=8, units='in', res=300)
grid.arrange(p.predleg,
             p.end, p.vlcm, p.mcr, p.pvm,
             p.ol, p.ast, p.rgl, p.neub2,
             p.grni, p.grnm, p.mcyp, p.madc,
             p.mklk, p.gabac, p.gabal, m.cr,
             layout_matrix=rbind(c(1,2,3,4,5),
                                 c(1,6,7,8,9),
                                 c(1,10,11,12,13),
                                 c(1,14,15,16,17)))
dev.off()


```